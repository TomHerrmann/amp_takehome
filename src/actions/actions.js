import {
  API_ERROR,
  CONTACTS_POPULATE,
  CONTACT_SELECT,
  CONTACT_UPDATE_CANCEL,
  CONTACT_UPDATE_SAVE,
} from '../constants/actionTypes';

export const apiError = (err) => ({
  type: API_ERROR,
  payload: err,
});

export const contactsPopulate = (contacts) => ({
  type: CONTACTS_POPULATE,
  payload: contacts,
});

export const contactSelect = (currentContact, contactIndex) => ({
  type: CONTACT_SELECT,
  payload: { currentContact, contactIndex },
});

export const contactUpdateCancel = () => ({
  type: CONTACT_UPDATE_CANCEL,
  payload: null,
});

export const contactUpdateSave = (formState) => ({
  type: CONTACT_UPDATE_SAVE,
  payload: formState,
});
