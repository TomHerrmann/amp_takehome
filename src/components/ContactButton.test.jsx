import React from 'react';
import { useDispatch } from 'react-redux';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ContactButtonm from './ContactButton.jsx';
import { contactSelect } from '../actions/actions';

jest.mock('react-redux');

describe('<ContactButton />', () => {
  let defaultProps;
  let dispatch;
  beforeEach(() => {
    defaultProps = {
      contact: {
        id: '9367',
        firstName: 'Brandy',
        lastName: 'Bartell',
        address: '42991 Beahan Parkway',
        city: 'Veumberg',
        country: 'Venezuela',
        phone: '586-723-4824',
        email: 'Marianne56@hotmail.com',
      },
      index: 0,
    };

    dispatch = jest.fn();
    useDispatch.mockReset();
    useDispatch.mockReturnValue(dispatch);
  });

  it('should render correctly', () => {
    const component = shallow(<ContactButtonm {...defaultProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should dispatch the correct action on button click', () => {
    const component = shallow(<ContactButtonm {...defaultProps} />);
    component.props().onClick();
    expect(dispatch).toHaveBeenCalledWith(
      contactSelect(defaultProps.contact, defaultProps.index)
    );
  });
});
