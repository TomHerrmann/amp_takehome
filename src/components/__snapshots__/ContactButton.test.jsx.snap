// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactButton /> should render correctly 1`] = `
<button
  className="contact-button"
  onClick={[Function]}
>
  Brandy Bartell
</button>
`;
